app/drivers/kscan/kscan_gpio_round_robin.c

```diff
@@ -0,0 +1,413 @@
+/*
+ * Copyright (c) 2020-2022 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include "debounce.h"
+
+#include <device.h>
+#include <devicetree.h>
+#include <drivers/gpio.h>
+#include <drivers/kscan.h>
+#include <kernel.h>
+#include <logging/log.h>
+#include <sys/__assert.h>
+#include <sys/util.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#define DT_DRV_COMPAT zmk_kscan_gpio_round_robin
+
+#define INST_LEN(n) DT_INST_PROP_LEN(n, gpios)
+#define INST_ROUND_ROBIN_LEN(n) (INST_LEN(n) * INST_LEN(n))
+
+#if CONFIG_ZMK_KSCAN_DEBOUNCE_PRESS_MS >= 0
+#define INST_DEBOUNCE_PRESS_MS(n) CONFIG_ZMK_KSCAN_DEBOUNCE_PRESS_MS
+#else
+#define INST_DEBOUNCE_PRESS_MS(n)                                                                  \
+    DT_INST_PROP_OR(n, debounce_period, DT_INST_PROP(n, debounce_press_ms))
+#endif
+
+#if CONFIG_ZMK_KSCAN_DEBOUNCE_RELEASE_MS >= 0
+#define INST_DEBOUNCE_RELEASE_MS(n) CONFIG_ZMK_KSCAN_DEBOUNCE_RELEASE_MS
+#else
+#define INST_DEBOUNCE_RELEASE_MS(n)                                                                \
+    DT_INST_PROP_OR(n, debounce_period, DT_INST_PROP(n, debounce_release_ms))
+#endif
+
+#define USE_POLLING IS_ENABLED(CONFIG_ZMK_KSCAN_ROUND_ROBIN_POLLING)
+#define USE_INTERRUPT (!USE_POLLING)
```

Comment:
You should be able to define a couple defines like

```
#define ANY_POLLING DT_INST_FOREACH(ADDS_ONE_IF_NO_INTERRUPT_GPIO) > 0
#define ANY_INTERRUPT DT_INST_FOREACH(ADDS_ONE_IF_HAS_INTERRUPT_GPIO) > 0
```

To use for the conditional *code* needed for each approach, and then similar "per instance" checks to decide what to do for a given instance.

Reply:
Didn't see this early enough, have made a change including both tracks, which would give the 'per instance' code anyway.

I'll take another look later.